client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -142,15 +142,16 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             final Map<String, Object> propsToAdd = new HashMap<>();
             // check the three types not allowed by AMQP 1.0 spec for application properties (list, map and array)
             for (final Map.Entry<String, ?> entry: properties.entrySet()) {
-                if (entry.getValue() instanceof List) {
-                    throw new IllegalArgumentException(String.format("Application property %s can't be a List", entry.getKey()));
-                } else if (entry.getValue() instanceof Map) {
-                    throw new IllegalArgumentException(String.format("Application property %s can't be a Map", entry.getKey()));
-                } else if (entry.getValue().getClass().isArray()) {
-                    throw new IllegalArgumentException(String.format("Application property %s can't be an Array", entry.getKey()));
-                } else {
-                    propsToAdd.put(entry.getKey(), entry.getValue());
+                if (entry.getValue() != null) {
```

Comment:
2 remarks:
- `entry.getValue().getClass().isArray()` was the code line I aimed at regarding being NP safe
- I wanted to (still) support null values in the message properties (since it was originally implemented to allow a null value for tenantId).
Do you have concerns with supporting null values in message properties? Thinking about it now it seems to have no different semantics than just not setting the key-value-pair at all, but it could be left to the client to decide about that IMHO.